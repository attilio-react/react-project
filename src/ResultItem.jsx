import React from 'react';

import {Label} from './Label.jsx';
import {Button} from './Button.jsx';

class ResultItem extends React.PureComponent {
  render () {
    const {item} = this.props;
    return <div>
                  <p><img src={item.imgSrc} /></p>
                  <p>
		     <Label text={item.releaseDate} />
		     &nbsp;
                     <Label text={item.genre} />
		  </p>
                  <p><Label text={item.title} /></p>
                  
           </div>;
  }
}


export {SearchByContainer}


