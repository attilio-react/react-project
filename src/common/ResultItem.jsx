import React from 'react';
import {Link} from "react-router-dom";

import {paths} from 'Common/Paths.jsx';

import {Label} from 'Common/Label.jsx';
import {Button} from 'Common/Button.jsx';

class ResultItem extends React.PureComponent {

 render () {
    const {item, clickCallback} = this.props,
          self = this;
    return <div style={{border: '1px solid black'}}><Link to={paths.DETAILS_PATH + item.id}>
                  <p><img src={item.poster_path} /></p>
                  <p>
		     <Label text={item.release_date} />
		     &nbsp;
                     <Label text={item.genres.join(', ')} />
		  </p>
                  <p><Label text={item.title} /></p>
                 </Link> 
           </div>;
  }
}


export {ResultItem}


