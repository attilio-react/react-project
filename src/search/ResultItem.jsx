import React from 'react';

import {Label} from 'Common/Label.jsx';
import {Button} from 'Common/Button.jsx';

class ResultItem extends React.PureComponent {
  render () {
    const {item} = this.props;
    return <div>
                  <p><img src={item.poster_path} /></p>
                  <p>
		     <Label text={item.release_date} />
		     &nbsp;
                     <Label text={item.genres.join(', ')} />
		  </p>
                  <p><Label text={item.title} /></p>
                  
           </div>;
  }
}


export {ResultItem}


