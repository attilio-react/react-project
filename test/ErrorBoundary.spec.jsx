import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';


import {ErrorBoundary} from '|ErrorBoundary.jsx';
import {ResultItem} from '|common/ResultItem.jsx';

describe('NoResult rendering', () => {
  it('Should render when there are no errors', () => {
    const 
        component = renderer.create(<ErrorBoundary><div>Test</div></ErrorBoundary>),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('Should render with errors', () => {
    const 
        component = renderer.create(<ErrorBoundary><ResultItem /></ErrorBoundary>),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});


