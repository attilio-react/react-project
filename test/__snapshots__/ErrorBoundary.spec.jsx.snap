// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResult rendering Should render when there are no errors 1`] = `
<div>
  Test
</div>
`;

exports[`NoResult rendering Should render with errors 1`] = `
<h1>
  An unexpected error occurred. Please try refershing the page.
</h1>
`;
